google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageImpl.java

```diff
@@ -567,7 +615,27 @@ public HmacKey createHmacKey(ServiceAccount serviceAccount, CreateHmacKeyOption.
 
   @Override
   public Page<HmacKeyMetadata> listHmacKeys(ListHmacKeysOption... options) {
-    return todo();
+    UnaryCallable<ListHmacKeysRequest, ListHmacKeysPagedResponse>
+        listBucketsRequestListHmacKeysPagedResponseUnaryCallable =
+            grpcStorageStub.listHmacKeysPagedCallable();
+    final Map<StorageRpc.Option, ?> optionsMap = StorageImpl.optionMap(options);
+    // TODO: Project is required?
```

Comment:
Project is required.

For the JSON implementation, we first attempt to read the supplied option for project_id, if that is null fall back to using the project id from options. We can do the same thing here.

https://github.com/googleapis/java-storage/blob/feat/grpc-storage/google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java#L1447-L1450